lib/url_utils.py

```diff
@@ -1,26 +1,152 @@
 import requests
 from bs4 import BeautifulSoup
-
 from pyspark.sql.functions import udf
+from lxml import html
 
 
-def extract_link_to_image(text: str) -> str:
+def try_extract_link_to_image(text: str) -> str:
+    """metoda dla wejściowego tekstu zwraca część tego tekst znajdującą się pomiędzy sekwencjami znaków 'src="' i "._";
+    metoda obsługuje wyjątki poprzez zwrócenie elemntu None"""
     try:
         first = 'src="'
-        last = "._V1"
+        last = "._"
         start = text.rindex(first) + len(first)
         end = text.rindex(last, start)
         return text[start:end]
     except ValueError:
         return None
 
 
-def get_link_to_imdb_image(actor_id: str) -> str:
+def try_get_link_to_imdb_image(actor_id: str) -> str:
+    """metoda przyjmuje jako argument id aktora ze strony IMDB i zwraca link do jego/jej zdjęcia za pomocą biblioteki
+    BeautifulSoup, korzystając z metody try_extract_link_to_image do ucięcia części linku odpowiadającej za
+    formatowanie"""
     url = f"https://www.imdb.com/name/{actor_id}/mediaindex"
     response = requests.get(url)
     soup = BeautifulSoup(response.content, "html.parser")
     image_info = str(soup.find("img", attrs={"class": "poster"}))
-    return extract_link_to_image(image_info)
+    return try_extract_link_to_image(image_info)
+
+
+udf_try_get_link_to_imdb_image = udf(try_get_link_to_imdb_image)
+
+
+def extract_link_to_image(text: str) -> str:
+    """metoda zwraca tekst znajdujący się pomiędzy sekwencjami znaków 'src="' oraz '._'"""
+    text = str(text)
```

Comment:
jeśli `text: str` to czemu 
```
text - str(text)
```
?

Reply:
Pewnie jakaś pozostałość po kopiowaniu kodu. Linijka została usunięta